File: Seccion 11 - Funciones en C++/Seccion_11_Python/25_ejercicio_19.py

import typing

def suma(n: int):
    if n == 1:
        return 1
    else:
        return n + suma(n - 1)

nElementos: int = 0

while nElementos <= 0:
    nElementos = int(input('Digite el número de elementos: '))
    if nElementos <= 0:
        print('\nEl número de elementos debe ser un entero positivo, no cero!!')

print('\nLa suma es: {}'.format(suma(nElementos)))

File: Seccion 11 - Funciones en C++/Seccion_11_Dart/25_ejercicio_19.dart

import 'dart:io';

int suma(int n) {
  if (n == 1) {
    return 1;
  } else {
    return n + suma(n - 1);
  }
}

void main() {

  int nElementos;

  do {
    stdout.writeln('Digite el número de elementos: ');
    nElementos = int.parse(stdin.readLineSync());
    if (nElementos <= 0) {
      print('\nEl número de elementos debe ser un entero positivo, no cero!!');
    }
  } while (nElementos <= 0);

  print('La suma es: ${suma(nElementos)}');
}